File: python/exercise_228.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        if len(nums) == 1:
            return [str(nums[0])]
        res = []
        currentValue = nums[0]
        rangeStart = 0 
        for i in range(len(nums)):
            if i == 0:
                continue
            if not (nums[i] == (nums[i-1]+1)):
                if rangeStart != i-1:
                    res.append(str(nums[rangeStart])+"->"+str(nums[i-1]))
                else:
                    res.append(str(nums[i-1]))
                rangeStart = i
            if i == len(nums)-1:
                print(rangeStart)
                if rangeStart == i:
                    res.append(str(nums[i]))
                else:
                    res.append(str(nums[rangeStart])+"->"+str(nums[i]))
        return res

File: java/arrays/exercise_228.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        if(nums.length == 0){
            return new ArrayList<>();
        }
        int start = nums[0];
        int count = 0;
        List<String> res = new ArrayList<>();
        for (int i=0;i<nums.length;i++){
            if(nums[i]>start+count){
                if (start == nums[i-1]){
                    res.add(String.valueOf(start));
                }else{
                    res.add(String.valueOf(start)+"->"+String.valueOf(nums[i-1]));
                }
                start = nums[i];
                count = 0;
            }
            count++;
        }
        if(start == nums[nums.length-1]){
            res.add(String.valueOf(start));
        }else{
            res.add(String.valueOf(start)+"->"+String.valueOf(nums[nums.length-1]));
        }
        return res;
    }
}